src/main/java/ch/njol/skript/expressions/ExprExplosionCause.java

```diff
@@ -63,8 +65,8 @@ public boolean isSingle() {
 	}
 	
 	@Override
-	public Class<? extends String> getReturnType() {
-		return String.class;
+	public Class<?> getReturnType() {
+		return Object.class; // Can't get event here and isCurrentEvent returns a list
```

Comment:
this will cause issues, you'll have to override the init method, do the check and call super like so:
```java
private boolean isEntity;

@Override
public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
  isEntity = getParser().isCurrentEvent(EntityExplodeEvent.class);
  if (!getParser().isCurrentEvent(BlockExplodeEvent.class) || !isEntity) {
    Skript.error("Cannot use the 'explosion cause' outside of an explode event");
    return false;
  }
  return super.init(exprs, matchedPattern, isDelayed, parseResult);
}
```

Reply:
`isEntity` will always be true in that code, the current event classes is always `EntityExplodeEvent.class` and `BlockExplodeEvent.class`.
(also, there's no super init method)

Reply:
I brain farted for a moment and thought it was a simple property expression for whatever reason. Goes to show I shouldn't write code before my morning coffee